pkg/virt-controller/watch/node.go

```diff
@@ -239,9 +239,26 @@ func (c *NodeController) alivePodsOnNode(nodeName string) ([]*v1.Pod, error) {
 		if controllerRef := controller.GetControllerOf(pod); !isControlledByVMI(controllerRef) {
 			continue
 		}
+
+		// Some pods get stuck in a pending Termination during shutdown
+		// due to virt-handler not being available to unmount container disk
+		// mount propagation. A pod with all containers terminated is not
+		// considered alive
+		allContainersTerminated := false
```

Comment:
If we have a final state  and there is no container status there, shouldn't we then consider that they are not running?

Reply:
yep, that's how the logic works. 

pods aren't added to the list of "alive" pods unless the pod is not finalized and all containers are not terminated.

```
                if !allContainersTerminated && phase != v1.PodFailed && phase != v1.PodSucceeded {
                        pods = append(pods, pod)
                        continue
                }
```

so if a pod is in a final state, it isn't considered alive regardless if there are no container status.